examples/common/chip-app-server/Server.cpp

```diff
@@ -61,23 +61,13 @@ class ServerCallback : public SecureSessionMgrDelegate
 {
 public:
     void OnMessageReceived(const PacketHeader & header, const PayloadHeader & payloadHeader,
-                           const Transport::PeerConnectionState * state, System::PacketBufferHandle buffer,
+                           SecureSessionHandle session, System::PacketBufferHandle buffer,
                            SecureSessionMgr * mgr) override
     {
-        const size_t data_len = buffer->DataLength();
-        char src_addr[PeerAddress::kMaxToStringSize];
-
         // as soon as a client connects, assume it is connected
         VerifyOrExit(!buffer.IsNull(), ChipLogProgress(AppServer, "Received data but couldn't process it..."));
         VerifyOrExit(header.GetSourceNodeId().HasValue(), ChipLogProgress(AppServer, "Unknown source for received message"));
-
-        VerifyOrExit(state->GetPeerNodeId() != kUndefinedNodeId, ChipLogProgress(AppServer, "Unknown source for received message"));
-
-        state->GetPeerAddress().ToString(src_addr, sizeof(src_addr));
-
-        ChipLogProgress(AppServer, "Packet received from %s: %zu bytes", src_addr, static_cast<size_t>(data_len));
-
-        HandleDataModelMessage(header, std::move(buffer), mgr);
```

Comment:
Maybe switch this code to reporting the node id instead of the source address (which it can get from the `SecureSessionHandle` , presumably, at least for a CASE session)?  Either way, keeping the "we got N bytes logging" would be useful.

Reply:
The member of the `SecureSessionHandle` is not decided yet. I don't want to expose the internal detail of `SecureSessionHandle` class.

I'm still considering what do we expose from session interface.

```c++
auto session = mSecureSessionManager->GetSession(handle /* an secure session handle */);
bool isPaseSession = session->IsPaseSession(); // or session->IsCaseSession() for case check.
auto nodeid = session->GetPeerNodeId();
auto localKey = session->GetLocalKeyId();
auto peerKey = session->GetPeerKeyId();
```

Reply:
That's fair, but we really will need a way to get the peer node ID for CASE sessions.  And we really should keep the length logging here.

Reply:
Good, I'll make it so.